manager/allocator/network.go

```diff
@@ -1039,8 +1044,8 @@ func (a *Allocator) procUnallocatedServices(ctx context.Context) {
 	nc := a.netCtx
 	var allocatedServices []*api.Service
 	for _, s := range nc.unallocatedServices {
-		if nc.nwkAllocator.ServiceNeedsAllocation(s) {
-			if err := a.allocateService(ctx, s); err != nil {
+		if !nc.nwkAllocator.IsServiceAllocated(s) {
```

Comment:
IsServiceAllocated() is missing in networkallocator.NetworkAllocator. Any ideas @fcrisciani